rt/rt.mjs

```diff
@@ -174,20 +179,39 @@ export class Engine {
     }
 
     jsFromScheme(ptr) {
-        return js_from_scheme(ptr);
+        switch (extract_tag(ptr)) {
+            case TAGS.string:
+                return this.schemeToString(ptr);
+            case TAGS.symbol:
+                if (this.carOf(ptr) != CONSTANTS.null) {
```

Comment:
I believe the logic is that if the car is a list of chars instead of a string, then it's a gensym.  In a new world where strings can be retagged as vectors, then I think the right thing would be to check if it's a vector instead.